File: src/main/java/com/zhongan/devpilot/completions/inline/AcceptDevPilotInlineCompletionAction.java
//
// Source code recreated from a .class file by IntelliJ IDEA
// (powered by FernFlower decompiler)
//

package com.zhongan.devpilot.completions.inline;

import com.intellij.codeInsight.hint.HintManagerImpl;
import com.intellij.openapi.actionSystem.DataContext;
import com.intellij.openapi.editor.Caret;
import com.intellij.openapi.editor.Editor;
import com.intellij.openapi.editor.actionSystem.EditorAction;
import com.intellij.openapi.editor.actionSystem.EditorWriteActionHandler;

public class AcceptDevPilotInlineCompletionAction extends EditorAction implements HintManagerImpl.ActionToIgnore, InlineCompletionAction {
    public static final String ACTION_ID = "AcceptDevPilotInlineCompletionAction";

    public AcceptDevPilotInlineCompletionAction() {
        super(new AcceptInlineCompletionHandler());
    }

    private static class AcceptInlineCompletionHandler extends EditorWriteActionHandler {
        private AcceptInlineCompletionHandler() {
        }

        public void executeWriteAction(Editor editor, Caret caret, DataContext dataContext) {
            CompletionPreview.getInstance(editor).applyPreview(caret != null ? caret : editor.getCaretModel().getCurrentCaret());
        }

        protected boolean isEnabledForCaret(Editor editor, Caret caret, DataContext dataContext) {
            return CompletionPreview.getInstance(editor) != null;
        }
    }
}

File: src/main/java/com/zhongan/devpilot/completions/common/inline/AcceptDevPilotInlineCompletionAction.kt
package com.zhongan.devpilot.completions.common.inline

import com.intellij.codeInsight.hint.HintManagerImpl.ActionToIgnore
import com.intellij.openapi.actionSystem.DataContext
import com.intellij.openapi.editor.Caret
import com.intellij.openapi.editor.Editor
import com.intellij.openapi.editor.actionSystem.EditorAction
import com.intellij.openapi.editor.actionSystem.EditorWriteActionHandler

object AcceptDevPilotInlineCompletionAction :
    EditorAction(AcceptInlineCompletionHandler()),
    ActionToIgnore,
        InlineCompletionAction {
    const val ACTION_ID = "AcceptDevPilotInlineCompletionAction"

    class AcceptInlineCompletionHandler : EditorWriteActionHandler() {
        override fun executeWriteAction(editor: Editor, caret: Caret?, dataContext: DataContext) {
            CompletionPreview.getInstance(editor)?.applyPreview(caret ?: editor.caretModel.currentCaret)
        }

        override fun isEnabledForCaret(
            editor: Editor,
            caret: Caret,
            dataContext: DataContext
        ): Boolean {
            return CompletionPreview.getInstance(editor) != null
        }
    }
}
